kie-internal/src/main/java/org/kie/internal/persistence/jpa/JPAKnowledgeService.java

```diff
@@ -162,7 +162,8 @@ private static synchronized KieStoreServices getJPAKnowledgeServiceProvider() {
     private static void loadProvider() {
         try {
             // we didn't find anything in properties so lets try and us reflection
-            Class<KieStoreServices> cls = (Class<KieStoreServices>) Class.forName( "org.drools.persistence.jpa.KnowledgeStoreServiceImpl" );
+            Class<KieStoreServices> cls = (Class<KieStoreServices>) Class.forName( System.getProperty( "org.kie.store.services.class", 
+                                                                                        "org.drools.persistence.jpa.KnowledgeStoreServiceImpl" ) );
             setJPAKnowledgeServiceProvider( cls.newInstance() );
         } catch ( Exception e ) {
             throw new RuntimeException( "Provider org.drools.persistence.jpa.KnowledgeStoreServiceImpl could not be set.",
```

Comment:
It seems the exception message should be updated as well, to print the actual class name (in case it was overridden by the property)?